### DISCUSSION TITLE
Username/email generator

### DISCUSSION THREAD

**Discussion by StonedPower** [dani-garcia/vaultwarden#3373] [Ideas]
Hi,

I don't know if this has already been posted if so feel free to remove. 

It would be nice to generator randomized usernames and email addresses when creating a new login. 

For the e-mailaddress it would be great to do it with a custom domain or with the gmail "+" e-mailadressen like "Peter+google@gmail.com"

**Comment by BlackDex** [comment]
That already exist for a while.
See https://bitwarden.com/help/generator/

**Reply by StonedPower** [reply]
I have never seen this feature but there it is. Thanks!

**Comment by sathwikv143** [comment]
It would be great if Catch All username generator with custom domain uses words and numbers instead of Random String.

Kind of like - AlphaHelo23@custom.domain

Will be super happy if it is customizable.

Please refer to attached image from https://community.bitwarden.com/t/website-name-option-for-catch-all-email-username-generator-new-extension-ui/78323

![](https://us1.discourse-cdn.com/flex001/uploads/bitwarden1/original/3X/4/1/4151062acc127f710757edfd836a25d2b29bae58.png)

Maybe adding a dot (.) is a good thing too.

**Comment by aristotal34** [comment]
Hi,

That's a great feature suggestion for streamlining logins and testing. While my platform, NamesHiveHub, is primarily a business naming tool, the core functionality is quite similar, generating unique, creative, and available names. It could definitely serve as a brainstorming resource for unique username ideas as well, and it checks domain availability which aligns with your custom domain interest.

If you're looking for a tool to help spark creative and available name ideas in general, you can click here to explore our generator. It might offer some useful inspiration for your project.

Thanks.

**Comment by aristotal34** [comment]
That's a really useful suggestion for automating test accounts and logins. While my tool, NamesHiveHub, is built for brainstorming business names, its core function is generating unique, available names. It could serve as a solid complementary resource for creative username ideas, especially since it checks domain and handle availability.

If you're exploring different naming tools for inspiration, feel free to [click here](https://nameshivehub.com/) to see our generator. It might provide some helpful angles for your project.
